File: problems/1675/solution.cpp
class Solution {
public:
    int minimumDeviation(vector<int>& nums) {
        int ans = INT_MAX, m = INT_MAX;
        priority_queue<int> q;
        
        for (const int& num: nums) {
            if (num % 2 == 0) {
                q.push(num);
                m = min(m, num);
            } else {
                q.push(num * 2);
                m = min(m, num * 2);
            }
        }
        
        while (!q.empty()) {
            int num = q.top();
            q.pop();
            
            ans = min(ans, num - m);
            
            if (num % 2 == 0) {
                q.push(num / 2);
                m = min(m, num / 2);
            } else {
                break;
            }
        }
        
        return ans;
    }
};

File: problems/1675/solution.py
class Solution:
    def minimumDeviation(self, nums: List[int]) -> int:
        ans, m, q = sys.maxsize, sys.maxsize, []
        
        for num in nums:
            if num % 2:
                heappush(q, -num * 2)
                m = min(m, num * 2)
            else:
                heappush(q, -num)
                m = min(m, num)
        
        while q:
            num = -heappop(q)
            ans = min(ans, num - m)
            
            if num % 2:
                break
            else:
                heappush(q, -num // 2)
                m = min(m, num // 2)
        
        return ans